### PR TITLE
Add python tests for handle_command in pywasmcross.py

### PR DESCRIPTION
**Pull request by rth**
This adds a mechanism for testing arguments conversion in `handle_command` within `pywasmcross.py`. An optional argument `pretend=False` is added to skip any OS-level operations. The tests are by no mean exhaustive, and are more intended to help contributors understand the expected behavior without running a build that would trigger some particular input command. New assertions could be added in the future as we run into more edge cases.

This also computes some code coverage in the `test-python` CI job. It's not very meaningful, but still gives an idea of how much code in `pyodide_build` and `src/pyodide.py` is covered by pure python tests without running build. The main idea is again to allow to fail as soon as possible. Current output can be found below,
```
pytest -v test/ -k 'not (firefox or chrome)' --cov=pyodide_build --cov=pyodide                      [pywasmcross-tests] 
==================================================================== test session starts ====================================================================
platform linux -- Python 3.7.0, pytest-3.7.4, py-1.6.0, pluggy-0.7.1 -- /home/rth/.miniconda3/envs/pyodide-env/bin/python
cachedir: .pytest_cache
rootdir: /home/rth/src/pyodide, inifile: setup.cfg
plugins: instafail-0.4.0, cov-2.6.0
collected 904 items / 902 deselected                                                                                                                        

test/pyodide_build/test_pywasmcross.py::test_handle_command PASSED                                                                                    [ 50%]
test/src/test_pyodide.py::test_find_imports PASSED                                                                                                    [100%]

----------- coverage: platform linux, python 3.7.0-final-0 -----------
Name                           Stmts   Miss  Cover
--------------------------------------------------
pyodide_build/__init__.py          1      0   100%
pyodide_build/__main__.py         14     14     0%
pyodide_build/_fixes.py           11      9    18%
pyodide_build/buildall.py         51     51     0%
pyodide_build/buildpkg.py        104    104     0%
pyodide_build/common.py           10      3    70%
pyodide_build/pywasmcross.py     139     81    42%
src/pyodide.py                    33     15    55%
--------------------------------------------------
TOTAL                            363    277    24%


========================================================= 2 passed, 902 deselected in 0.27 seconds ==========================================================
```

### CODE REVIEW

File: test/pyodide_build/test_pywasmcross.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+import sys
+
+sys.path.append(str(Path(__file__).parents[2]))
+
+from pyodide_build.pywasmcross import handle_command  # noqa: E402
+
+
+def _args_wrapper(func):
+    """Convert function to take as input / return a string instead of a
+    list of arguments
```
Comment (rth) [inline test/pyodide_build/test_pywasmcross.py]: This is mostly for convenience to allow asserting,
```py
handle_command_wrap('gcc -shared -c test.o -o test.so', args)
```
instead of,
```py
handle_command(['gcc', '-shared', '-c', 'test.o', '-o', 'test.so'], args)
```
but I can remove it if you find it not helpful.

File: pyodide_build/pywasmcross.py [hunk 2]
```diff
@@ -105,7 +105,7 @@ def capture_compile(args):
         sys.exit(result.returncode)
 
 
-def handle_command(line, args):
+def handle_command(line, args, pretend=False):
```
Comment (mdboom) [inline pyodide_build/pywasmcross.py]: I usually see this called `dryrun`.  And maybe prefix it with `_` so that it's clear it's a testing detail?

**Comment by rth** [comment]
Thanks. Renamed the parameter to `dryrun`. Didn't add an underscore in the name, as I think for someone running this function manually, not executing the transformed command is a valid use case. Added a docstring and a doctest to improve the documentation about how `handle_command` works.

**Comment by rth** [comment]
The CI looks fine, please let me know @mdboom if you have other comments on this PR. Thanks!

**Comment by rth** [comment]
Thanks @mdboom !
